### PR TITLE
feat: ACNA-2585 - library configuration & networking

### PR DESCRIPTION
**Pull request by shazron**
## How Has This Been Tested?

- npm test
- e2e test on local storage-next server

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by moritzraho — COMMENTED** [review]
recommendations,
- validate regions: oneOf amer, apac and emea
- default region: amer

wdyt?

**Comment by shazron** [comment]
> recommendations,
> 
> * validate regions: oneOf amer, apac and emea
> * default region: amer
> 
> wdyt?

SGTM!

File: lib/constants.js [hunk 1]
```diff
@@ -11,16 +11,26 @@ governing permissions and limitations under the License.
...
+    'apac',
+    'emea'
+  ]
 } = process.env
 
 const ADOBE_STATE_STORE_ENDPOINT = {
-  [PROD_ENV]: ADOBE_STATE_STORE_ENDPOINT_PROD,
-  [STAGE_ENV]: ADOBE_STATE_STORE_ENDPOINT_STAGE
+  [PROD_ENV]: isInternalToAdobeRuntime() ? ADOBE_STATE_STORE_ENDPOINT_PROD_INTERNAL : ADOBE_STATE_STORE_ENDPOINT_PROD,
+  [STAGE_ENV]: isInternalToAdobeRuntime() ? ADOBE_STATE_STORE_ENDPOINT_STAGE_INTERNAL : ADOBE_STATE_STORE_ENDPOINT_STAGE
```
Comment (moritzraho) [inline lib/constants.js:33]: we have an internal issue for stage, the internal endpoint will not work but neither will the corp one in runtime. so let's keep it like this just writing this comment for visibility.

File: test/AdobeState.test.js [hunk 2]
```diff
@@ -376,7 +379,19 @@ describe('private methods', () => {
...
+      const region = 'apac'
+      const env = PROD_ENV
+      getCliEnv.mockReturnValue(env)
+
+      // need to instantiate a new store, when env changes
+      const store = await AdobeState.init({ ...fakeCredentials, region })
+
+      const url = store.createRequestUrl()
+      expect(url).toEqual(`https://${region}.${myConstants.ADOBE_STATE_STORE_ENDPOINT[env]}/${API_VERSION}/containers/${fakeCredentials.namespace}`)
     })
```
Comment (moritzraho) [inline test/AdobeState.test.js:395]: missing a test for invalid/not-supported region?

**Review by moritzraho — APPROVED** [review]
nice work! (just added one small comment regarding testing)

File: e2e/.env.example [hunk 3]
```diff
@@ -2,3 +2,4 @@ TEST_NAMESPACE_1=
 TEST_AUTH_1=
 TEST_NAMESPACE_2=
 TEST_AUTH_2=
+# ADOBE_STATE_STORE_ENDPOINT_PROD
```
Comment (MichaelGoberling) [inline e2e/.env.example:5]: nit: Remove?

**Review by MichaelGoberling — APPROVED** [review]

File: test/AdobeState.test.js (hunk 2 above)
Comment (shazron) [reply test/AdobeState.test.js:395]: fixed via https://github.com/adobe/aio-lib-state/pull/138/commits/06e4ba3ba421260f1d2964b2d98e06ff52c715c0

File: doc/api.md [hunk 4]
```diff
@@ -167,7 +167,7 @@ AdobeState put options
 
 | Name | Type | Description |
 | --- | --- | --- |
-| ttl | <code>number</code> | time-to-live for key-value pair in seconds, defaults to 24 hours (86400s). Set to < 0 for no expiry. A value of 0 sets default. |
+| ttl | <code>number</code> | time-to-live for key-value pair in seconds, defaults to 24 hours (86400s). Set to < 0 for max ttl of one year. A value of 0 sets default. |
```
Comment (moritzraho) [inline doc/api.md:170]: nice

File: doc/api.md [hunk 5]
```diff
@@ -179,7 +179,7 @@ AdobeState get return object
 
 | Name | Type | Description |
 | --- | --- | --- |
-| expiration | <code>string</code> \| <code>null</code> | ISO date string of expiration time for the key-value pair, if the ttl is infinite expiration=null |
+| expiration | <code>number</code> \| <code>null</code> | time in milliseconds since epoch, for the key-value pair expiration=null |
```
Comment (moritzraho) [inline doc/api.md:182]: hmm, I think we should still expose it like in v2 for now?

**Review by moritzraho — COMMENTED** [review]
I would argue to not change the way we expose the expiration to users in the library (keep iso), wdyt?

File: doc/api.md (hunk 5 above)
Comment (shazron) [reply doc/api.md:182]: ok, I'll revert back to ISO string.

**Comment by shazron** [comment]
> I would argue to not change the way we expose the expiration to users in the library (keep iso), wdyt?

makes sense, ISO string is unambiguous, and better, I think

File: doc/api.md (hunk 5 above)
Comment (shazron) [reply doc/api.md]: fixed in https://github.com/adobe/aio-lib-state/pull/138/commits/d5fc278a03bfeae79dd1d23eaeeb3ba8917f4e14

**Review by moritzraho — APPROVED** [review]

**Pull request merged**